app/models/spree/gateway/amazon.rb

```diff
@@ -183,6 +183,21 @@ def void(response_code, gateway_options)
       return ActiveMerchant::Billing::Response.new(true, "Success", response)
     end
 
+    def cancel(response_code)
+      payment = Spree::Payment.find_by!(response_code: response_code)
+      order = payment.order
+      load_amazon_mws(payment.source.order_reference)
+      capture_id = order.amazon_transaction.capture_id
+
+      if capture_id.nil?
+        response = @mws.cancel(order.amazon_transaction.order_reference)
+      else
+        response = @mws.refund(response_code, order.number, payment.credit_allowed, payment.currency)
+      end
+
+      return ActiveMerchant::Billing::Response.new(true, "#{order.number}-cancel", response)
```

Comment:
FYI regarding the `"#{order.number}-cancel"` here:  I had requested something like that [here](https://github.com/boomerdigital/solidus_amazon_payments/pull/30#r75179056), but I was referencing the call to `@mws.refund`, not for the `Response.new`.  I don't think the second parameter to `Response.new` is really used for anything in Solidus.